rtnl/addr.go

```diff
@@ -0,0 +1,102 @@
+// github.com/jsimonetti/rtnetlink/rtnl
+package rtnl
+
+import (
+	"github.com/jsimonetti/rtnetlink"
+	"golang.org/x/sys/unix"
+	"net"
+)
+
+func AddrAdd(conn *rtnetlink.Conn, ifc *net.Interface, addr *net.IPNet) error {
+	af := addrFamily(addr.IP)
+	prefixlen, _ := addr.Mask.Size()
+	bcastIP := broadcastAddr(addr)
+	if bcastIP == nil {
+		bcastIP = net.IP{}
+	}
```

Comment:
Is it appropriate to add checks in here to verify the IP isn't already set on the interface?

Reply:
I don't think so. If you try to do that, you will receive a 'file exists' error from rtnetlink.

Reply:
I think this is something that should be checked by the caller

Reply:
something like this: https://github.com/insomniacslk/dhcp/blob/master/netboot/netconf.go#L223

Reply:
I'd imagine that'd be a common chunk of boilerplate code that'd need to be written each time someone uses this versus handling it here. Not enough to be a blocker, but seems like it'd be beneficial from a caller perspective.

Reply:
I would argue that the user wants the function to do what it says it does, and get the error, whatever it is. But the scenario you suggest is not uncommon, and I would rather add an `IsRouteExists` helper instead of trying to be smart in the function, which may not be what the user wants